File: services/src/main/java/com/liferay/mobile/sdk/v7/commentmanagerjsonws/CommentManagerJSONWSService.java
package com.liferay.mobile.sdk.v7.commentmanagerjsonws;

import com.liferay.mobile.sdk.Call;
import com.liferay.mobile.sdk.annotation.Param;
import com.liferay.mobile.sdk.annotation.Path;
import com.liferay.mobile.sdk.http.Response;
import java.lang.Boolean;
import java.lang.Integer;
import java.lang.Long;
import java.lang.String;
import org.json.JSONArray;

@Path("/comment.commentmanagerjsonws")
public interface CommentManagerJSONWSService {
  @Path("/add-comment")
  Call<Long> addComment(@Param(name = "groupId") long groupId, @Param(name = "className") String className, @Param(name = "classPK") long classPK, @Param(name = "body") String body);

  @Path("/delete-comment")
  Call<Response> deleteComment(@Param(name = "commentId") long commentId);

  @Path("/get-comments")
  Call<JSONArray> getComments(@Param(name = "groupId") long groupId, @Param(name = "className") String className, @Param(name = "classPK") long classPK, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-comments-count")
  Call<Integer> getCommentsCount(@Param(name = "groupId") long groupId, @Param(name = "className") String className, @Param(name = "classPK") long classPK);

  @Path("/has-discussion")
  Call<Boolean> hasDiscussion(@Param(name = "groupId") long groupId, @Param(name = "className") String className, @Param(name = "classPK") long classPK);

  @Path("/subscribe-discussion")
  Call<Response> subscribeDiscussion(@Param(name = "groupId") long groupId, @Param(name = "className") String className, @Param(name = "classPK") long classPK);

  @Path("/unsubscribe-discussion")
  Call<Response> unsubscribeDiscussion(@Param(name = "groupId") long groupId, @Param(name = "className") String className, @Param(name = "classPK") long classPK);

  @Path("/update-comment")
  Call<Long> updateComment(@Param(name = "className") String className, @Param(name = "classPK") long classPK, @Param(name = "commentId") long commentId, @Param(name = "subject") String subject, @Param(name = "body") String body);
}

File: windows/Liferay.SDK/Service/V72/Commentmanagerjsonws/CommentmanagerjsonwsService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V72.Commentmanagerjsonws
{
	public class CommentmanagerjsonwsService : ServiceBase
	{
		public CommentmanagerjsonwsService(ISession session)
			: base(session)
		{
		}

		public async Task DeleteCommentAsync(long commentId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commentId", commentId);

			var _command = new JsonObject()
			{
				{ "/comment.commentmanagerjsonws/delete-comment", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<long> GetCommentsCountAsync(long groupId, string className, long classPK)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("className", className);
			_parameters.Add("classPK", classPK);

			var _command = new JsonObject()
			{
				{ "/comment.commentmanagerjsonws/get-comments-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<bool> HasDiscussionAsync(long groupId, string className, long classPK)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("className", className);
			_parameters.Add("classPK", classPK);

			var _command = new JsonObject()
			{
				{ "/comment.commentmanagerjsonws/has-discussion", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (bool)_obj;
		}

		public async Task SubscribeDiscussionAsync(long groupId, string className, long classPK)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("className", className);
			_parameters.Add("classPK", classPK);

			var _command = new JsonObject()
			{
				{ "/comment.commentmanagerjsonws/subscribe-discussion", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task UnsubscribeDiscussionAsync(long groupId, string className, long classPK)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("className", className);
			_parameters.Add("classPK", classPK);

			var _command = new JsonObject()
			{
				{ "/comment.commentmanagerjsonws/unsubscribe-discussion", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<long> UpdateCommentAsync(string className, long classPK, long commentId, string subject, string body)
		{
			var _parameters = new JsonObject();

			_parameters.Add("className", className);
			_parameters.Add("classPK", classPK);
			_parameters.Add("commentId", commentId);
			_parameters.Add("subject", subject);
			_parameters.Add("body", body);

			var _command = new JsonObject()
			{
				{ "/comment.commentmanagerjsonws/update-comment", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> AddCommentAsync(long groupId, string className, long classPK, string body)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("className", className);
			_parameters.Add("classPK", classPK);
			_parameters.Add("body", body);

			var _command = new JsonObject()
			{
				{ "/comment.commentmanagerjsonws/add-comment", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommentsAsync(long groupId, string className, long classPK, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("className", className);
			_parameters.Add("classPK", classPK);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/comment.commentmanagerjsonws/get-comments", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommentsAsync(long commentId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commentId", commentId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/comment.commentmanagerjsonws/get-comments", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}
	}
}